modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx

```diff
@@ -187,6 +191,31 @@ export default function ViewObjectDefinitions({
 		return url;
 	};
 
+	useEffect(() => {
+		if (objectFoldersRequestInfo?.items.length > 1) {
+			const itemsActions = [...objectDefinitionsFDSActionDropdownItems];
+			itemsActions.push({
+				data: {
+					id: 'moveObjectDefinition',
+					method: 'update',
+					permissionKey: 'update',
+				},
+				href: null,
+				icon: 'move-folder',
+				label: 'Move',
+				target: null,
+				type: 'item',
+			});
+			setUpdatedFDSItemsActions(itemsActions);
+		}
+		else if (objectFoldersRequestInfo?.items.length === 1) {
+			setUpdatedFDSItemsActions(objectDefinitionsFDSActionDropdownItems);
```

Comment:
When we add a folder, there is no page reload, so the first `if` evaluates to `true` and we do our thing.

When we delete a folder, the whole page reloads and the front-end state resets to initial. In the case there is only one folder (uncategorized) and this `else if` evaluates to `true`, it will do nothing because `objectDefinitionsFDSActionDropdownItems` is already the initial state.  I think this `else if` can be removed.I think this `else if` can be removed. What do you think?

Reply:
hmm i think you are right! i will test removing this code to see if it works, thanks!